File: solutions/kotlin/resistor-color/1/src/main/kotlin/ResistorColor.kt
object ResistorColor {

    private val colorList = listOf(
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    )

    fun colorCode(input: String): Int {
        return colorList.indexOf(input)
    }

    fun colors(): List<String> {
        return colorList
    }
}

File: solutions/julia/resistor-color/1/resistor-color.jl
const COLOR_LIST = [
    "black", "brown", "red", "orange", "yellow", 
    "green", "blue", "violet", "grey", "white"
]

function colorcode(color)

    return findfirst(==(color), COLOR_LIST) - 1
end

function colors()
    # Возвращаем весь список цветов
    return COLOR_LIST
end
